Meta/CMake/utils.cmake

```diff
@@ -247,3 +249,21 @@ function(extract_path dest_dir zip_path source_path dest_path)
         file(ARCHIVE_EXTRACT INPUT "${zip_path}" DESTINATION "${dest_dir}" PATTERNS "${source_path}")
     endif()
 endfunction()
+
+function(depend_on_clang_plugin target_name plugin_name)
+    add_dependencies(${target_name} ${plugin_name}Target)
+    target_compile_options(${target_name} PRIVATE -fplugin=${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_LIBDIR}/lib${plugin_name}.so)
+endfunction()
+
+function(compile_with_clang_plugins target_name)
```

Comment:
rather than this, can we do:

```cmake
add_library(JSClangPlugin INTERFACE)
add_library(GenericClangPlugin INTERFACE)
if (ENABLE_CLANG_PLUGINS AND CMAKE_CXX_COMPILER_ID MATCHES "Clang$")
   depend_on_clang_plugin(JSClangPlugin LibJSGCClangPlugin) # need to modify target_compile_options to INTERFACE instead in this func
   depend_on_clang_plugin(GenericClangPlugin LambdaCaptureClangPlugin)
endif()
```
? 

Then we could just add the interface library here to the target_link_libraries in serenity_lib/serenity_bin and the ones that want GC checking... or add the JS one as a public dep of LibJS?

`target_link_libraries(LibJS PUBLIC JSClangPlugin)`

If we make it a public dep, then we don't need to edit all the app/lib cmake lists (except for LibJS's)

This is a similar pattern to what I did for the NoCoverage target.

Reply:
This seems to only work for the lagom build. I've tried to fix the CI failure but haven't been able to. I'm confused why an interface target would be producing a linker flag since from what I've read, interface targets don't necessarily have any sources. And the fact that this failure is happening without `ENABLE_CLANG_PLUGINS` begin `ON` is also confusing (since the interface targets are pretty much empty when that's the case).

Reply:
@ADKaster Do you have any ideas on how to fix this?